src/ui/public/parse_query/__tests__/parse_query.js

```diff
@@ -0,0 +1,48 @@
+import expect from 'expect.js';
+import { formatQuery, parseQuery } from 'ui/parse_query';
+
+describe('parseQuery', function () {
+  it('should treat an empty string as a match_all', function () {
+    expect(parseQuery('')).to.eql({ match_all: {} });
+  });
+
+  it('should treat input that does not start with { as a query string', function () {
+    expect(parseQuery('foo')).to.eql({ query_string: { query: 'foo' } });
+    expect(parseQuery('400')).to.eql({ query_string: { query: '400' } });
+    expect(parseQuery('true')).to.eql({ query_string: { query: 'true' } });
+  });
+
+  it('should parse valid JSON', function () {
+    expect(parseQuery('{}')).to.eql({});
+    expect(parseQuery('{a:b}')).to.eql({ query_string: { query: '{a:b}' } });
+  });
+
+  it('should present undefined as empty string', function () {
```

Comment:
Super minor, but from here on the rest of the tests should probably be in a `formatQuery` describe block.